feat(web): add tutorial for staging wp site (#3120)
scaleway/docs-content: tutorials/create-deploy-staging-site-cpanel/index.mdx

Previously, setting up a staging site involved a convoluted and time-intensive process of data export, database creation, file copying, and configuration adjustment, followed by a reverse procedure for deployment.
This complexity discouraged many WordPress users from using staging sites, favoring live edits instead. However, for WordPress sites hosted on [Scaleway Web Hosting](https://www.scaleway.com/en/web-hosting/), powered by cPanel servers, the setup is simplified to just a few clicks.

Creating a staging site for your WordPress website is a smart move, allowing you to test changes before making them live. Use cases for staging sites include:
* Testing new landing pages.
* Developing or modifying themes.
* Implementing code changes to enhance functionality.
* Evaluating updates for compatibility issues.

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- A [Web Hosting](/managed-services/webhosting/) account

## Creating a WordPress staging site

> 1. Click **Web Hosting** in the **Managed Services** section of the [console](https://console.scaleway.com/) side menu. The **Web Hosting** overview page displays.

Reviewer: ```suggestion

1. Click **Web Hosting** in the **Managed Services** section of the [console](https://console.scaleway.com/) side menu. The **Web Hosting** overview page displays.
```

2. Click <Icon name="seeMore" /> or the Web Hosting domain name you want to configure. The **Hosting information** page displays.
3. Click **Control panel dashboard** to open the cPanel interface in a new tab.
4. Log in to your cPanel account and navigate to the **Domains** section. Click **WP Toolkit** to display the WP Toolkit dashboard.
   <Lightbox src="scaleway-cpanel-wp-toolkit.webp" alt="" size="medium" />
5. From the WP Toolkit dashboard, locate the site you want to work with and click on it to access detailed options.

> 6. find and click on the Clone tool within the detailed view. The clone interface displays.

Reviewer: ```suggestion
6. Find and click the **Clone** tool within the detailed view. The clone interface displays.
```

   <Lightbox src="scaleway-cpanel-wp-clone.webp" alt="" size="medium" />
7. Designate the subdomain and path for your staging site in the clone interface.
   <Message type="note">
     You can either use an existing domain or let WP Toolkit create a new one for you. Additionally, you can customize the database name if needed.
   </Message>
8. You can add labels to distinguish staging sites from live ones. Click **Add Label** and choose an appropriate label from the drop-down menu. (This step is optional but recommended.)
  <Lightbox src="scaleway-cpanel-wp-label.webp" alt="" size="medium" />

Once the cloning process is complete, your staging site will be ready. You can access it through the main WP Toolkit interface alongside your other sites.
Use your staging site to test changes, such as new features, themes, or code edits. Ensure everything works as expected before proceeding.

## Deploying changes to production

> After testing, it's time to push the changes to your main site.

Reviewer: ```suggestion

After testing, it's time to push the changes to your main site.
```

1. Go to the staging site's WP Toolkit dashboard and click the **Copy Data** tool.
 <Lightbox src="scaleway-cpanel-wp-copy.webp" alt="" size="medium" />

> 2. select your production site from the drop-down menu in the Copy Data interface.

Reviewer: ```suggestion
2. Select your production site from the drop-down menu in the **Copy Data** interface.
```

3. Choose which aspects of the site to transfer, including files, database, and specific tables.
4. Choose whether to overwrite or delete production files affected by changes made on the staging site.
5. Click **Start** to proceed with the copying process.
<Message type="tip">
 WP Toolkit automatically creates a restore point before copying data, providing a safety net if anything goes wrong during deployment.
</Message>

## Conclusion

> With WP Toolkit's clone and copy features, setting up a staging site and deploying changes becomes quick and straightforward. You can create a staging site in seconds and quickly push tested changes to your production site.

Reviewer: ```suggestion

With WP Toolkit's clone and copy features, setting up a staging site and deploying changes becomes quick and straightforward. You can create a staging site in seconds and quickly push tested changes to your production site.
```

---
meta:

>   title: Create and deploy a WordPress staging site on Scaleway Web Hosting

Reviewer 2: ```suggestion
  title: Create and deploy a WordPress staging website on Scaleway Web Hosting
```

>   description: Scaleway's WordPress hosting simplifies staging site setup to facilitate testing of changes before going live, optimizing workflow and ensuring compatibility.

Reviewer 2: ```suggestion
  description: Scaleway's WordPress hosting simplifies staging website setup to facilitate testing of changes before going live, optimizing workflow and ensuring compatibility.
```

content:

>   h1: Create and deploy a WordPress staging site on Scaleway Web Hosting
>   paragraph: Scaleway's WordPress hosting simplifies staging site setup to facilitate testing of changes before going live, optimizing workflow and ensuring compatibility.

Reviewer 2: ```suggestion
  h1: Create and deploy a WordPress staging website on Scaleway Web Hosting
  paragraph: Scaleway's WordPress hosting simplifies staging website setup to facilitate testing of changes before going live, optimizing workflow and ensuring compatibility.
```

tags: webhosting cpanel wordpress staging production
dates:
  validation: 2024-04-26
  posted: 2024-04-26
categories:
  - webhosting
---

> A staging site mirrors your production site, containing identical data and files while remaining publicly inaccessible.
> Changes made on the staging site are isolated until being transferred to the production site after thorough checks from design and development teams. Operating in the same environment as the production environment, it serves as an ideal testing ground.
> 
> Previously, setting up a staging site involved a convoluted and time-intensive process of data export, database creation, file copying, and configuration adjustment, followed by a reverse procedure for deployment.
> This complexity discouraged many WordPress users from using staging sites, favoring live edits instead. However, for WordPress sites hosted on [Scaleway Web Hosting](https://www.scaleway.com/en/web-hosting/), powered by cPanel servers, the setup is simplified to just a few clicks.
> 
> Creating a staging site for your WordPress website is a smart move, allowing you to test changes before making them live. Use cases for staging sites include:

Reviewer 2: ```suggestion
A staging website mirrors your production website, containing identical data and files while remaining publicly inaccessible.
Changes made on the staging website are isolated until being transferred to the production website after thorough checks from design and development teams. Operating in the same environment as the production environment, it serves as an ideal testing ground.

Previously, setting up a staging website involved a convoluted and time-intensive process of data export, database creation, file copying, and configuration adjustment, followed by a reverse procedure for deployment.
This complexity discouraged many WordPress users from using staging websites, favoring live edits instead. However, for WordPress websites hosted on [Scaleway Web Hosting](https://www.scaleway.com/en/web-hosting/), powered by cPanel servers, the setup is simplified to just a few clicks.

Creating a staging website for your WordPress website is a smart move, allowing you to test changes before making them live. Use cases for staging websites include:
```

* Testing new landing pages.
* Developing or modifying themes.
* Implementing code changes to enhance functionality.
* Evaluating updates for compatibility issues.

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- A [Web Hosting](/managed-services/webhosting/) account

> ## Creating a WordPress staging site

Reviewer 2: ```suggestion
## Creating a WordPress staging website
```

1. Click **Web Hosting** in the **Managed Services** section of the [console](https://console.scaleway.com/) side menu. The **Web Hosting** overview page displays.
2. Click <Icon name="seeMore" /> or the Web Hosting domain name you want to configure. The **Hosting information** page displays.
3. Click **Control panel dashboard** to open the cPanel interface in a new tab.
4. Log in to your cPanel account and navigate to the **Domains** section. Click **WP Toolkit** to display the WP Toolkit dashboard.
   <Lightbox src="scaleway-cpanel-wp-toolkit.webp" alt="" size="medium" />

> 5. From the WP Toolkit dashboard, locate the site you want to work with and click on it to access detailed options.
> 6. Find and click the **Clone** tool within the detailed view. The clone interface displays.
>    <Lightbox src="scaleway-cpanel-wp-clone.webp" alt="" size="medium" />
> 7. Designate the subdomain and path for your staging site in the clone interface.
>    <Message type="note">
>      You can either use an existing domain or let WP Toolkit create a new one for you. Additionally, you can customize the database name if needed.
>    </Message>
> 8. You can add labels to distinguish staging sites from live ones. Click **Add Label** and choose an appropriate label from the drop-down menu. (This step is optional but recommended.)
>   <Lightbox src="scaleway-cpanel-wp-label.webp" alt="" size="medium" />
> 
> Once the cloning process is complete, your staging site will be ready. You can access it through the main WP Toolkit interface alongside your other sites.
> Use your staging site to test changes, such as new features, themes, or code edits. Ensure everything works as expected before proceeding.
> 
> ## Deploying changes to production
> 
> After testing, it's time to push the changes to your main site.

Reviewer 2: ```suggestion
5. From the WP Toolkit dashboard, locate the website you want to work with and click on it to access detailed options.
6. Find and click the **Clone** tool within the detailed view. The clone interface displays.
   <Lightbox src="scaleway-cpanel-wp-clone.webp" alt="" size="medium" />
7. Designate the subdomain and path for your staging website in the clone interface.
   <Message type="note">
     You can either use an existing domain or let WP Toolkit create a new one for you. Additionally, you can customize the database name if needed.
   </Message>
8. You can add labels to distinguish staging websites from live ones. Click **Add Label** and choose an appropriate label from the drop-down menu. (This step is optional but recommended.)
  <Lightbox src="scaleway-cpanel-wp-label.webp" alt="" size="medium" />

Once the cloning process is complete, your staging website will be ready. You can access it through the main WP Toolkit interface alongside your other websites.
Use your staging website to test changes, such as new features, themes, or code edits. Ensure everything works as expected before proceeding.

## Deploying changes to production

After testing, it's time to push the changes to your main website.
```

> 1. Go to the staging site's WP Toolkit dashboard and click the **Copy Data** tool.
>  <Lightbox src="scaleway-cpanel-wp-copy.webp" alt="" size="medium" />
> 2. Select your production site from the drop-down menu in the **Copy Data** interface.
> 3. Choose which aspects of the site to transfer, including files, database, and specific tables.
> 4. Choose whether to overwrite or delete production files affected by changes made on the staging site.

Reviewer 2: ```suggestion
1. Go to the staging website's WP Toolkit dashboard and click the **Copy Data** tool.
 <Lightbox src="scaleway-cpanel-wp-copy.webp" alt="" size="medium" />
2. Select your production website from the drop-down menu in the **Copy Data** interface.
3. Choose which aspects of the website to transfer, including files, database, and specific tables.
4. Choose whether to overwrite or delete production files affected by changes made on the staging website.
```

5. Click **Start** to proceed with the copying process.
<Message type="tip">
 WP Toolkit automatically creates a restore point before copying data, providing a safety net if anything goes wrong during deployment.
</Message>

## Conclusion

> With WP Toolkit's clone and copy features, setting up a staging site and deploying changes becomes quick and straightforward. You can create a staging site in seconds and quickly push tested changes to your production site.
> This streamlined process ensures faster, safer, and more reliable WordPress workflows, crucial for agencies and businesses managing multiple WordPress sites.

Reviewer 2: ```suggestion
With WP Toolkit's clone and copy features, setting up a staging website and deploying changes becomes quick and straightforward. You can create a staging website in seconds and quickly push tested changes to your production website.
This streamlined process ensures faster, safer, and more reliable WordPress workflows, crucial for agencies and businesses managing multiple WordPress websites.
```
